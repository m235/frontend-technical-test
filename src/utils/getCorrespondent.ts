import type { Conversation } from '@/types/conversation'

export const getCorrespondent = (userId: number, conversation: Conversation) => {
  switch (userId) {
    case conversation.senderId:
      return conversation.recipientNickname
    case conversation.recipientId:
      return conversation.senderNickname
    default:
      throw new Error('You are nor the recipient nor the sender')
  }
}
