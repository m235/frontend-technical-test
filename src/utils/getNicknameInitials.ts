export const getNicknameInitials = (nickname: string) =>
  nickname
    .split(' ')
    .map((string) => string.trim().charAt(0))
    .slice(0, 2)
    .join('')
