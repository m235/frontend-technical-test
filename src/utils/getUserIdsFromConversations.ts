import type { Conversation } from '@/types/conversation'

export const getUserIdsFromConversations = (conversations?: Conversation[]) =>
  Array.from<number>(new Set(conversations?.reduce((acc, curr) => [...acc, curr.recipientId, curr.senderId], []) ?? []))
