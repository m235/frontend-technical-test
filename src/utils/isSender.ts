import type { Conversation } from '@/types/conversation'

export const isSender = (userId: number, conversation: Conversation) => userId === conversation.senderId
