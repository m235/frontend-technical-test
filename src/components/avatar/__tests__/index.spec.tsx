import { render } from '@/tests/utils'

import Avatar from '@/components/avatar'

describe('components:avatar', () => {
  it('should render default correctly', () => {
    const { container } = render(<Avatar>Children</Avatar>)

    expect(container).toMatchSnapshot()
  })
  it('should render small correctly', () => {
    const { container } = render(<Avatar size="small">Children</Avatar>)

    expect(container).toMatchSnapshot()
  })
  it('should contain an img', () => {
    const { getByTestId } = render(
      <Avatar>
        <img data-testid="avatar" alt="fake" />
      </Avatar>
    )

    expect(getByTestId('avatar')).toBeTruthy()
  })
  it('should contain a simple text', () => {
    const { getByText } = render(<Avatar>Children</Avatar>)
    expect(getByText('C').textContent).toEqual('C')
  })
})
