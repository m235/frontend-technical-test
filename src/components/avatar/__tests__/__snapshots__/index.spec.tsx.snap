// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:avatar should render default correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ fCfHbp"
  >
    C
  </div>
</div>
`;

exports[`components:avatar should render small correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ knfGZE"
  >
    C
  </div>
</div>
`;
